superset/views/core.py

```diff
@@ -2777,6 +2777,13 @@ def _get_sqllab_tabs(user_id: Optional[int]) -> Dict[str, Any]:
     @expose("/sqllab/", methods=["GET", "POST"])
     def sqllab(self) -> FlaskResponse:
         """SQL Editor"""
+        if not (
+            security_manager.is_admin()
+            or "sql_lab" in (role.name for role in security_manager.get_user_roles())
+        ):
+            flash(__("You do not have access to SQL Lab"), "danger")
+            return redirect("/")
```

Comment:
This probably makes more sense to fix on security manager, where all roles are created

Reply:
Good idea, I'll move it over 👍